packages/components/src/components/data-list/data-list.tsx

```diff
@@ -0,0 +1,244 @@
+import classNames from 'classnames';
+import React from 'react';
+import { TransitionGroup } from 'react-transition-group';
+import {
+    List as _List,
+    CellMeasurer as _CellMeasurer,
+    CellMeasurerCache,
+    CellMeasurerProps,
+    AutoSizer as _AutoSizer,
+    type AutoSizerProps,
+    ScrollParams,
+    ListProps,
+    ListRowProps,
+} from 'react-virtualized';
+import { isMobile, isDesktop } from '@deriv/shared';
+import DataListCell from './data-list-cell';
+import DataListRow from './data-list-row';
+import ThemedScrollbars from '../themed-scrollbars';
+import { MeasuredCellParent } from 'react-virtualized/dist/es/CellMeasurer';
+
+const List = _List as unknown as React.FC<ListProps>;
+const AutoSizer = _AutoSizer as unknown as React.FC<AutoSizerProps>;
+const CellMeasurer = _CellMeasurer as unknown as React.FC<CellMeasurerProps>;
+type CellType = { Cell: typeof DataListCell };
+export type TRowRenderer = (params: { row: any; is_footer?: boolean; measure?: () => void }) => React.ReactNode;
+type DataListProps = {
+    className?: string;
+    data_source: any;
+    footer?: React.ReactNode;
+    getRowAction?: (row: React.ReactNode) => string;
+    getRowSize?: (params: any) => number;
+    keyMapper?: (row: any) => string;
+    onRowsRendered?: () => void;
+    onScroll?: (ev: ScrollParams) => void;
+    passthrough?: {
+        root_store: any;
+        WS: Record<string, any>;
+    };
+    row_gap?: number;
+    setListRef?: (ref: MeasuredCellParent) => void;
+    rowRenderer: TRowRenderer;
+    children?: React.ReactNode;
+    overscanRowCount?: number;
+    Cell?: CellType;
+};
+type GetContentType = { measure?: () => void | undefined };
+
+const DataList = ({
+    children,
+    className,
+    data_source,
+    footer,
+    getRowSize,
+    keyMapper,
+    onRowsRendered,
+    onScroll,
+    setListRef,
+    overscanRowCount,
+    rowRenderer: rowRendererProp,
+    row_gap,
+    getRowAction,
+    passthrough,
+}: DataListProps) => {
+    const [is_loading, setLoading] = React.useState(true);
+    const [is_scrolling, setIsScrolling] = React.useState(false);
+    const [scroll_top, setScrollTop] = React.useState(0);
+
+    const cache = React.useRef<CellMeasurerCache>();
+    const list_ref = React.useRef<MeasuredCellParent | null>(null);
```

Comment:
Not Important

```suggestion
    const list_ref = React.useRef<MeasuredCellParent>(null);
```

Reply:
for defining refs without default values it should be like this.